### PR TITLE
provide required schema for basic template

### PR DESCRIPTION
**Pull request by grokspawn**
**Description of the change:**
This standardizes all templates on formal schema (#1331), and lays groundwork for standardization of input (#1332) to support #1334 following a roadmap listed in #1330.

IMPORTANT NOTE!

Working on this, it made a lot of sense for `opm` to own the conversion to a template from FBC, since it controls/supports the templates' definitions.  So this PR introduces a new `opm alpha convert-template` command which explicitly only provides support for the basic template. 

Solves #1332 

**Motivation for the change:**
Eliminating the need for users (pipelines, authors, etc.) to be aware of which supported template type is being processed. 

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: alpha/template/basic/basic.go
Comment (grokspawn) [reply alpha/template/basic/basic.go:86]: Once we get some more use/feedback/maturity in the interface, the plan is to eventually promote the catalog templates out of alpha and into pkg.

File: cmd/opm/alpha/convert-template/convert.go
Comment (everettraven) [inline cmd/opm/alpha/convert-template/convert.go:44]:
```suggestion
			case "yaml", "json":
				converter.OutputFormat = output
```

File: cmd/opm/alpha/template/cmd.go
Comment (everettraven) [inline cmd/opm/alpha/template/cmd.go:17]: Remove?

File: cmd/opm/alpha/template/cmd.go
Comment (everettraven) [inline cmd/opm/alpha/template/cmd.go:21]: Remove?

File: cmd/opm/internal/util/util.go
Comment (everettraven) [inline cmd/opm/internal/util/util.go:71]: Nit: This is an internal function, and I know it was mostly a move, but the implementation only ever looks at the first argument. It seems reasonable to just make this a single string argument instead of a slice of strings
```suggestion
func OpenFileOrStdin(cmd *cobra.Command, path string) (io.ReadCloser, string, error) {
```

**Review by everettraven — APPROVED** [review]
Saw a few small things, but nothing that I think is worth holding this up over if you feel they aren't worth addressing

File: cmd/opm/alpha/convert-template/convert.go
Comment (grokspawn) [reply cmd/opm/alpha/convert-template/convert.go:44]: Reasoning?  When I wrote it I found (for e.g. [here](https://go.dev/wiki/Switch)) no idiomatic argument for either approach.

File: cmd/opm/internal/util/util.go
Comment (grokspawn) [reply cmd/opm/internal/util/util.go:71]: This has bugged me too, but it does actually check for [empty args](https://github.com/operator-framework/operator-registry/pull/1335/files#diff-9373b2d6ffc7be864852e489772ff2062a0a2359ff9f3be9f574ba26d69b8fedR72) as a condition for reading from stdin instead.
We could make it pass just the input path and have each caller test for empty args (and pass in an empty string) but then the functions utility is significantly reduced.

File: cmd/opm/alpha/template/cmd.go
Comment (grokspawn) [reply cmd/opm/alpha/template/cmd.go:17]: A planned follow-up in https://github.com/operator-framework/operator-registry/issues/1330 will use these.  I was initially doing it as part of this PR, but realized that it would balloon this PR and likely impact the speed of its review.  If you don't feel strongly, I'd just leave these since for some time the commands would be hidden-but-callable.

File: cmd/opm/internal/util/util.go
Comment (everettraven) [reply cmd/opm/internal/util/util.go:71]: I don't feel super strongly one way or the other, and saw the empty args check but figured an empty string would work equivalently. This shouldn't block this PR if you'd rather keep it as is.

File: cmd/opm/alpha/convert-template/convert.go
Comment (perdasilva) [reply cmd/opm/alpha/convert-template/convert.go:44]: I was gonna say, if this is possible, that's nice =D I can never remember fallthrough - well not exactly true, diarrhea in German is durchfall. So, I have that mnemonic XD

Comment (everettraven) [reply cmd/opm/alpha/convert-template/convert.go:44]: IMO using the `fallthrough` directive:
- increases mental complexity for a reader. They have to first understand what the fallthrough directive does and then build the mental model of "If the yaml case is hit it will fallthrough and do whatever the case underneath of it is, which is the json case and therefore will set the OutputFormat"
- `fallthrough` means execute the logic of the case underneath and could result in an easier introduction of bugs. What happens if someone adds an additional case below the `yaml` case that slips through and completely changes the behavior when `yaml` is specified as the output format?

Instead, using the OR syntax for a switch case is:
- Easier to reason about and build a mental model. Simply looking at the case statement tells me that either the value `yaml` or `json` executes the exact same code.
- Less prone to introducing bugs inadvertently as there is a well defined execution path that is not dependent on the ordering of case statements

**Review by perdasilva — APPROVED** [review]
lgtm - though I'd also suggest a someone more across the subject to ensure edge cases, etc. don't fall through the cracks.

File: cmd/opm/alpha/convert-template/convert.go
Comment (grokspawn) [reply cmd/opm/alpha/convert-template/convert.go]: I guess it's a question of subjective contexts here.  I infinitely prefer the explicit fallthrough idiom over for e.g. C's implicit (and problematic) fallthrough.  
And similarly, I prefer to have a single evaluated condition examined per case, based on experience with other languages.  
Like I said... context. 

Not a significant factor for me so I made the changes to unblock the review.

(Also, eww Per. 😉)

**Review by everettraven — APPROVED** [review]

**Pull request merged**
